src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,230 @@
+        Topics currentNetworkProxy = currentdeviceConfiguration.getNetworkProxyNamespace();
+        Topics newNetworkProxy = deploymentConfiguration.getNetworkProxyNamespace();
+        return !Topics.compareChildren(currentNetworkProxy, newNetworkProxy);
+    }
+
+    private boolean awsRegionHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentAwsRegion = currentdeviceConfiguration.getAWSRegion();
+        Topic newAwsRegion = deploymentConfiguration.getAWSRegion();
+        return !Topic.compareValue(currentAwsRegion, newAwsRegion);
+    }
+
+    private boolean mqttHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topics currentMqtt = currentdeviceConfiguration.getMQTTNamespace();
+        Topics newMqtt = deploymentConfiguration.getMQTTNamespace();
+        return !Topics.compareChildren(currentMqtt, newMqtt);
+    }
+
+    private boolean iotDataEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentIotDataEndpoint = currentdeviceConfiguration.getIotDataEndpoint();
+        Topic newIotDataEndpoint = deploymentConfiguration.getIotDataEndpoint();
+        return !Topic.compareValue(currentIotDataEndpoint, newIotDataEndpoint);
+    }
+
+    private boolean greengrassDataPlaneEndpointHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        // Defaults to empty string topic
+        Topic currentGreengrassDataPlaneEndpoint = currentdeviceConfiguration.getGGDataEndpoint();
+        Topic newGreengrassDataPlaneEndpoint = deploymentConfiguration.getGGDataEndpoint();
+        return !Topic.compareValue(currentGreengrassDataPlaneEndpoint, newGreengrassDataPlaneEndpoint);
+    }
+
+    private boolean greengrassDataPlanePortHasChanged(DeviceConfiguration currentdeviceConfiguration) {
+        Topic currentGreengrassDataPlanePort = currentdeviceConfiguration.getGreengrassDataPlanePort();
+        Topic newGreengrassDataPlanePort = deploymentConfiguration.getGreengrassDataPlanePort();
+        return !Topic.compareValue(currentGreengrassDataPlanePort, newGreengrassDataPlanePort);
+    }
+
+    /**
+     * Creates an MQTT client and checks if the device can connect to AWS.
+     *
+     * @param totallyCompleteFuture Future will be updated if validation fails
+     */
+    public boolean mqttClientCanConnect(CompletableFuture<DeploymentResult> totallyCompleteFuture) {
+        logger.atDebug().log("Checking MQTT client can connect");
+
+        String clientId = Coerce.toString(deploymentConfiguration.getThingName()) + clientIdSuffix;
+        try (AwsIotMqttConnectionBuilder builder = mqttClient.createMqttConnectionBuilder(deploymentConfiguration,
+                securityService, null);
+                MqttClientConnection connection = builder.withClientId(clientId).build()) {
```

Comment:
properly cleanup after yourself. You're making a new connection and never stopping and closing it.